examples/chart/teleport-cluster/templates/deployment.yaml

```diff
@@ -230,9 +231,9 @@ spec:
         configMap:
           name: {{ .Release.Name }}
       - name: "data"
-        {{- if eq .Values.chartMode "standalone" }}
+        {{- if and (.Values.persistence.enabled) (or (eq .Values.chartMode "standalone") (eq .Values.chartMode "custom")) }}
```

Comment:
If I'm reading this right, I think we may now have a different backwards compatibility problem to before...

`custom` mode is now fine and can optionally use `persistence.enabled`, but if someone is using the legacy `standalone` mode (i.e. where storage previously always defaulted to enabled), then they won't be setting `persistence.enabled: true` and their current PVC (and thus Teleport state) would vanish when they upgrade the chart.

I think the way it needs to work is like this:
```yaml
chartMode: custom
persistence:
  enabled: true
# PVC enabled

chartMode: custom
# PVC disabled

chartMode: standalone
persistence:
  enabled: true
# PVC enabled

chartMode: standalone
# PVC still enabled
```

Reply:
Well caught! Will find a way to handle that

Reply:
> Well caught! Will find a way to handle that

Actually I don't think there is an issue as the default `persistence.enabled: true` is always used by the chart, regardless of whether `standalone:` section is specified or not. The only way for `chartMode: standalone` users to lose data would be to either remove `persistence.enabled` field or set `persistence.enabled: false` from the chart original `values.yaml`.

Please note that directly editing/replacing a chart `values.yaml` (e.g.: pasting the content of an older chart `values.yaml` to the latest one) can be regarded as bad practise since default values might then be missing. `helm upgrade -f override.yaml ...` or `helm upgrade --set ...` is the correct and safest approach.

Of course don't hesitate to tell me if I misinterpreted your question or missed something, in fact your question made me realize another edge case related to persistenVolumeClaim size (fixed [here](https://github.com/gravitational/teleport/pull/11368/commits/371f557701c367a387fe1161ce71e32dcade1465)).

Reply:
Got it. I agree with your assessment and would hope most people are using the chart correctly i.e. allowing the default values to apply, and then overriding individual specific values on the command line with `-f values.yaml`. 

If someone has incorrectly copy/pasted an older version of the entire default values file and are then setting each thing there deliberately, I don't think this is a use-case we should be expected to support. Even then, they should have no value set for `persistence` which will mean they default to `persistence.enabled: true` and everything will work anyway.